### PR TITLE
make menu only ever load once

### PR DESCRIPTION
**Pull request by marcgear**
## Description

We now will only request the menu if it has not been loaded.  Server side requests will make the request to the platform for the menu data and then populate the redux store.  If that hasn't happened, for any reason (menu wasn't included/needed in the SSR) then it'll happen client side.  

It also prevents any unnecessary rerenders from happening client-side through use of shouldComponentUpdate()

## Related Issue

@shiftcommerce/trendy-golf#131

### Checklist

- [x] Appropriate commenting on functions/components (JSDoc format).
- [ ] All SHIFT library versions bumped and published (utilised in this PR).
- [x] Unit tests added/amended.
- [x] Integration tests added/amended.
- [x] Environment variables supplied (.env.example also updated)

### How has/can this been tested?

- [x] Serverside rendering
- [x] Clientside rendering

Verified this works by loading trendy and reference site using this code and everything works fine, and no client side getMenu calls are made.

### Dependencies Review
None

### CODE REVIEW

File: packages/reference/cypress/integration/search/price-filter.js [hunk 1]
```diff
@@ -27,7 +27,6 @@ describe('Search', () => {
       }).as('algoliaSearch')
 
       // Wait for page to load
-      cy.wait('@getMenus')
```
Comment (cbarratt) [inline packages/reference/cypress/integration/search/price-filter.js]: I think I added this as a global stub in Cypress inside of `global-stubs.js`, I'd be intrigued to see if we can remove it from there and everything passes 🤔

Comment (marcgear) [reply packages/reference/cypress/integration/search/price-filter.js]: worth seeing - i'll give it a go.  Technically this request can still get made, so we maybe shouldn't remove the stub altogether, but I can certianly see if it passes without

Comment (marcgear) [reply packages/reference/cypress/integration/search/price-filter.js]: yup - passes without that stub \o/

Comment (cbarratt) [reply packages/reference/cypress/integration/search/price-filter.js]: Nice, was curious more than anything :D as you say leave it in it won't harm anyhow!

File: packages/shift-react-components/src/components/navigation/navbar.js [hunk 2]
```diff
@@ -1,6 +1,10 @@
 // Libraries
 import React, { Component } from 'react'
 import t from 'typy'
+import { connect } from 'react-redux'
+import PropTypes from 'prop-types'
+
+import isEqual from 'lodash/isEqual'
```
Comment (Azfletch) [inline packages/shift-react-components/src/components/navigation/navbar.js]: Do we need to use lodash here?

Comment (Azfletch) [reply packages/shift-react-components/src/components/navigation/navbar.js]: Just looked at alternatives to lodash, not much available. Seems you're probably going about this in the most performant way. So forget my previous comment, i'm happy with this.

Comment (marcgear) [reply packages/shift-react-components/src/components/navigation/navbar.js]: No, but I didn't want to write my own deep equality checker ;)

**Pull request merged**